Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -205,6 +222,114 @@ ObjectNode::~ObjectNode()
     return true;
 }
 
+// DeterminePreBuildDynamicDependenciesNeedToBuild
+//------------------------------------------------------------------------------
+/*virtual*/ bool ObjectNode::DeterminePreBuildDynamicDependenciesNeedToBuild(bool forceClean) const
+{
+	if( GetPreBuildPreprocessor() == nullptr )
+		return false;
+
+	if ( forceClean ||
```

Comment:
Currently we only re-build dynamic dependencies if the source file or the preprocessor changes, since the assumption is that the preprocessor only gathers dependencies from the single input file. If this is not the case perhaps we should be checking if any of the static dependencies have changed?